dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -472,32 +473,33 @@ public CloseableClient build() {
 
             InetSocketAddress socksProxyAddress = getSocksProxyAddress(conf);
             SSLSocketFactory rawSocketFactory = conf.sslSocketFactory();
-            SSLConnectionSocketFactory sslSocketFactory =
-                    new SSLConnectionSocketFactory(
-                            MetricRegistries.instrument(conf.taggedMetricRegistry(), rawSocketFactory, name),
-                            TlsProtocols.get(),
-                            supportedCipherSuites(CipherSuites.allCipherSuites(), rawSocketFactory, name),
-                            new InstrumentedHostnameVerifier(
-                                    new DefaultHostnameVerifier(), name, conf.taggedMetricRegistry())) {
-                        @Override
-                        public Socket createSocket(final HttpContext context) throws IOException {
-                            return socksProxyAddress == null
-                                    ? super.createSocket(context)
-                                    : new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
-                        }
-                    };
+            Supplier<Socket> simpleSocketCreator = socksProxyAddress == null
+                    ? () -> new Socket(Proxy.NO_PROXY)
+                    : () -> new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
```

Comment:
Factored out rather than duplicating because we want identical behavior between the two